src/NServiceBus.Testing.Tests/HandlerTests.cs

```diff
@@ -194,6 +202,14 @@ public void ShouldPassExpectPublishWhenPublishing()
         public void ShouldFailExpectNotPublishWhenPublishing()
         {
             Assert.Throws<ExpectationException>(() => Test.Handler<PublishingHandler<Publish1>>()
+                .ExpectNotPublish<Publish1>()
+                .OnMessage<TestMessage>());
+        }
+
+        [Test]
+        public void ShouldFailExpectNotPublishWithCheckWhenPublishing()
+        {
+            Assert.Throws<ExpectationException>(() => Test.Handler<PublishingHandler<Publish1>>()
                 .ExpectNotPublish<Publish1>(m => true)
```

Comment:
Maybe this is a non-issue, but when I saw `m => true` I first thought:

> Hmmm... but the default check always returns true. 

Then I realized this should demonstrate the usage of the non-null check function. After that I'm asking myself if it would be better to actually check something inside the check function. Thoughts?